slurm_ops_manager/mpi.py

```diff
@@ -0,0 +1,53 @@
+"""MPI Class."""
+
+import logging
+import shlex
+import subprocess
+
+from ops.framework import Object, StoredState
+from slurm_ops_manager.utils import operating_system
+
+
+logger = logging.getLogger()
+
+
+class MPI(Object):
+    """Responsible for MPI operations."""
+
+    _stored = StoredState()
+
+    def __init__(self, parent, key):
+        """Initialize class."""
+        super().__init__(parent, key)
+
+        self._operating_system = operating_system()
+
+    def install(self):
+        """Install MPI (mpich)."""
+
+        logger.debug("#### Installing MPI (mpich)")
+
+        cmds = {"ubuntu": "apt-get install --yes mpich",
+                "centos": "yum install --assumeyes mpich-3.2 mpich-3.2-devel"}
```

Comment:
Could you also pin the version installed on Ubuntu? Otherwise we might not have the same version on different OSes.

What are the dependencies pulled on each OS? Does this also give the dev packages on Ubuntu?

Reply:
Done! It installs a few dependencies, like gcc/g++ compiler. In both centos and Ubuntu the dev packages are installed.